pkg/templates/loader/terraform_loader.go

```diff
@@ -205,12 +205,24 @@ func (l *TerraformLoader) applyMissingConfig(generated, customized *openapi3.Sch
 		}
 
 		// Extensions.
-		if len(in.Value.Extensions) != 0 {
-			if len(v.Value.Extensions) == 0 {
-				s.Properties[n].Value.Extensions = make(map[string]any)
-			}
-			s.Properties[n].Value.Extensions[openapi.ExtOriginal] = in.Value.Extensions[openapi.ExtOriginal]
+		ext := openapi.NewExt(v.Value.Extensions)
+		ext.SetOriginal(in.Value.Extensions[openapi.ExtOriginal])
+
+		order := openapi.GetUIOrder(v.Value.Extensions)
+		genOrder := openapi.GetUIOrder(in.Value.Extensions)
+
+		if order == 0 && genOrder != 0 {
+			ext.SetUIOrder(genOrder)
 		}
+
+		colSpan := openapi.GetUIColSpan(v.Value.Extensions)
+		genColSpan := openapi.GetUIOrder(in.Value.Extensions)
```

Comment:
Should we define the UIExtension structure so that it's easier and clearer for parsing and maintaining?

Reply:
updated